Weave compare trace docs 1904 (#2169)
wandb/docs: weave/guides/tools/comparison.mdx

description: "Visually compare and diff code, traces, prompts, models, and configurations"
---

The W&B Weave Comparison feature allows you to visually compare and diff code, traces, prompts, models, and model configurations. You can compare two objects side-by-side or analyze a larger set of objects to identify differences, patterns, and trends.

This guide covers the steps to start a comparison and the available actions to tailor your comparison view, including baseline comparisons, numeric diff formatting, and more.

## Access the Comparison view

> 1. In the sidebar, select the type of object to compare (for example **Traces** or **Models**).

Reviewer: ```suggestion
1. In the sidebar, select the type of object to compare (for example, **Traces** or **Models**).
```

2. Select the objects that you want to compare. The selection method varies depending on the type of object you are comparing:
   - For **Traces**, select traces to compare by checking the checkboxes for the appropriate rows in the Trace column.
   - For objects such as **Models**, navigate to the model Versions page and check the checkboxes next to the versions that you want to compare.

> 3. Click **Compare** to open the Comparison view. At the top of the page, the comparison bar contains interactive call ID tokens (for example, `f02b`, `4a98`) that you can drag, reorder, or remove to change the comparison.

Reviewer: ```suggestion
3. Select **Compare** to open the Comparison view. At the top of the page, the comparison bar contains interactive call ID tokens (for example, `f02b`, `4a98`) that you can drag, reorder, or remove to change the comparison.
```

### Set a baseline

By default, each object in the Comparison view is compared to the object to its left. However, you can set an object as the _baseline_, which means that all objects will be compared to the leftmost object in the view.

To set an object as the baseline:
1. In the comparison bar, hold the pointer over the object that you want to set as the baseline.
2. Click the ellipses (<Icon icon="ellipsis-vertical" iconType="solid" />) to the right of the ID.
   ![Comparison view showing the **Make baseline** option for the selected object in the comparison bar.](/weave/guides/tools/imgs/comparison-2objs-baseline.png)
3. In the list, select **Make baseline**. The UI refreshes with the baseline object in the leftmost position in the comparison bar and `Baseline` next to the ID.
    ![Comparison view showing the selected baseline object in the leftmost position of the comparison bar.](/weave/guides/tools/imgs/comparison-2objs-baseline-set.png)

### Remove a baseline

> To remove an object as baseline:

Reviewer: ```suggestion
To remove an object as a baseline:
```

This option is only available when comparing 3 or more objects. You can also [set](#set-a-baseline) or [remove an existing baseline by clicking the 3 dots to the right of the ID](#remove-a-baseline).

To perform a baseline comparison with 3 or more objects:

> 1. In the right hand corner of either the Side-by-side or Unified Comparison view, click the list. Depending on your current view configuration, the list is either titled **Compare with previous** or **Compare with baseline**.

Reviewer: ```suggestion
1. In the right-hand corner of either the Side-by-side or Unified Comparison view, click the list. Depending on your current view configuration, the list is either titled **Compare with previous** or **Compare with baseline**.
```

This option is only available when comparing 3 or more objects.

When comparing 3 or more objects, you can compare a single object to its adjacent object or a baseline. This changes the Comparison table view so that the view is identical to a two-object comparison. 

To compare a pair of objects from a multi-object comparison:

> 1. In the comparison bar, find the ID that you want to compare. If you have a baseline set, the comparison will be made to the baseline object. Otherwise, it will compare to the prior ID (the ID to the left) in the list of selected IDs.

Reviewer: I'm finding this line a little vague. I'm assuming you're supposed to find and select the ID of the object you want to compare, right? 

Also, the last two sentences use the future tense and the passive voice, which we usually try to avoid. I relent on passive voice sometimes, though, because it can be easier to just plainly say what something is in passive voice rather than doing grammar gymnastics just for the sake of having active voice.

> 1. In the comparison bar, find the ID that you want to compare. If you have a baseline set, the comparison will be made to the baseline object. Otherwise, it will compare to the prior ID (the ID to the left) in the list of selected IDs.

Author: ```suggestion
1. In the comparison bar, select the ID you want to compare. If you have set a baseline, the view compares the selected object to the baseline. Otherwise, it compares the object to the ID immediately to its left in the comparison bar.
```
